runtime-sdk/src/crypto/signature/secp384r1.rs

```diff
@@ -0,0 +1,144 @@
+//! Secp384r1 signatures.
+use digest::{consts::U48, core_api::BlockSizeUser, Digest, FixedOutput, FixedOutputReset};
+use p384::{
+    self,
+    ecdsa::{
+        self,
+        signature::{DigestSigner as _, DigestVerifier, Signer as _, Verifier as _},
+    },
+};
+
+use crate::crypto::signature::{Error, Signature};
+
+/// A Secp384r1 public key (in compressed form).
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct PublicKey(p384::EncodedPoint);
+
+impl PublicKey {
+    /// Return a byte representation of this public key.
+    pub fn as_bytes(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+
+    /// Construct a public key from a slice of bytes.
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, Error> {
+        let ep = p384::EncodedPoint::from_bytes(bytes).map_err(|_| Error::MalformedPublicKey)?;
+        if !ep.is_compressed() {
+            // This should never happen due to the size check above.
+            return Err(Error::MalformedPublicKey);
+        }
+        Ok(PublicKey(ep))
+    }
+
+    /// Verify a signature.
+    pub fn verify(
+        &self,
+        context: &[u8],
+        message: &[u8],
+        signature: &Signature,
+    ) -> Result<(), Error> {
+        self.verify_digest(
```

Comment:
I would change Secp256r1 and Secp256k1 a bit, just to be consistent:
- Remove `digest` from `verify`.
- Update `Signer`.
```
    fn sign(&self, context: &[u8], message: &[u8]) -> Result<Signature, Error> {
        let digest = Sha512_256::new()
            .chain_update(context)
            .chain_update(message);
            ...
    }
```